### PR TITLE
Add 3.8 upgrade test job

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/katello/3.8-upgrade.groovy [hunk 1]
```diff
@@ -0,0 +1,4 @@
+def playBookVars() {
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/katello/3.8-upgrade.groovy]: Where is this file used?

Comment (zjhuntin) [reply puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/katello/3.8-upgrade.groovy]: I think it was meant to be added to the job file, but has not yet, both of `3.8-upgrade` and `3.8` have not been added to `foreman-katello-upgrade-3.8-test`.

Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/katello/3.8-upgrade.groovy]: Correct, I missed updating the naming. That has been updated

**Comment by ehelms** [comment]
This requires https://github.com/theforeman/forklift/pull/814

File: puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-katello-upgrade-3.8-test.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+- job:
+    name: 'foreman-katello-upgrade-nightly-test'
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-katello-upgrade-3.8-test.yml]:
```
ERROR:jenkins_jobs.parser:Duplicate entry found in '/home/travis/build/theforeman/foreman-infra/puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-katello-upgrade-nightly-test.yml: 'foreman-katello-upgrade-nightly-test' already defined
```

surely, this should be `foreman-katello-upgrade-3.8-test`

**Pull request merged**